pkg/controller/replicaset/replica_set.go

```diff
@@ -635,9 +644,6 @@ func (rsc *ReplicaSetController) claimPods(rs *apps.ReplicaSet, selector labels.
 		if err != nil {
 			return nil, err
 		}
-		if fresh.UID != rs.UID {
```

Comment:
why is this removed? it seems important?

Reply:
Good catch.  At one point I was checking UIDs elsewhere and backed out those changes.  This looks like it was errantly removed as part of that cleanup.